### ISSUE TITLE
Task sourcesJar uses this output of task generateAvroJava without declaring an explicit or implicit dependency

### ISSUE THREAD

**Issue by thorin** [davidmc24/gradle-avro-plugin#167]
**Prerequisites**

* [x] Are you running the latest version of the plugin? (Check [releases](https://github.com/davidmc24/gradle-avro-plugin/releases))
* [x] Are you running a supported version of Gradle? (Check the [README](https://github.com/davidmc24/gradle-avro-plugin/blob/master/README.md))
* [x] Are you running a supported version of Apache Avro? (Check the [README](https://github.com/davidmc24/gradle-avro-plugin/blob/master/README.md))
* [x] Are you running a supported version of Java? (Check the [README](https://github.com/davidmc24/gradle-avro-plugin/blob/master/README.md))
* [x] Did you check to see if an [issue](https://github.com/davidmc24/gradle-avro-plugin/issues) has already been submitted?
* [x] Are you reporting to the correct repository?  If your schema doesn't work with the Apache Avro CLI tool either, it's not a problem with this plugin.  Running your file through the `CLIComparisonTest` in the sample project under the `test-project` directory can help diagnose this.
* [x] Did you perform a cursory search?

**Describe the bug**

Gradle 7 when running the task `sourcesJar` detects that this task uses outputs from `generateAvroJavo` without any implicit or explicit dependency between the two tasks.

```
:generateAvroJava	
Execution optimizations have been disabled for task ':generateAvroJava' to ensure correctness due to the following reasons:	
  - Gradle detected a problem with the following location: '.../build/generated-main-avro-java'. Reason: Task ':sourcesJar' uses this output of task ':generateAvroJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.	
```

**To Reproduce**

File structure:
```
./gradlew.bat
./gradle
./gradle/wrapper
./gradle/wrapper/gradle-wrapper.jar
./gradle/wrapper/gradle-wrapper.properties
./build.gradle
./src
./src/main
./src/main/avro
./src/main/avro/Sample.asc
./gradlew
```

build.gradle:
```groovy
plugins {
  id 'java-library'
  id 'com.github.davidmc24.gradle.plugin.avro' version '1.2.0'
}

repositories {
   mavenCentral()
}

dependencies {
    api "org.apache.avro:avro:1.10.2"
}

java {
  withSourcesJar()
}
```

./src/main/avro/Sample.asc:
```
{
  "type" : "record",
  "name" : "DefaultContext",
  "namespace" : "test.avro",
  "fields" : [
    {
      "name" : "id",
      "type" : [ "null", "string" ]
    },{
      "name" : "message",
      "type" : [ "null", "string" ]
    }
  ]
}
```

**Expected behavior**

Sources jar should have an implicit dependency from the **generateAvroJava** output.

**Environment (please complete the following information):**
 - Gradle Version: 7.1.1
 - Apache Avro Version: 1.10.2
 - Gradle-Avro Plugin Version: 1.2.0
 - Java Version: 1.8.0_282
 - OS: Ubuntu focal

**Additional context**

A workaround for this is to include the following snippet of code:

```groovy
tasks.named('sourcesJar').configure {
    dependsOn generateAvroJava
}
```

**Comment by davidmc24** [comment]
Thanks for the bug report. I'll try to investigate tonight.

**Comment by davidmc24** [comment]
I've been able to reproduce the behavior.  I'll need to dig into Gradle's source a bit to figure out how it's determining what sources to use, to figure out if the problem is on my side or Gradle's.

**Comment by davidmc24** [comment]
I believe I have a working fix in master.  I'll take another look on Friday, and if it still looks good to me, I'll push out a release.

**Comment by thorin** [comment]
Many thanks David.

**Comment by davidmc24** [comment]
Included in the upcoming 1.2.1 release (which I'm working on publishing right now).
